product_document_domain/models/product_template.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _compute_product_document_count(self):
```

Comment:
Kind of similar here. You could call super method and then reduce the records with domain ('res_model', '=', 'product.product'), ('res_id', 'in', template.product_variant_ids.ids). Has it been completely completely overriden for any specific reason?